spec/models/ccms/submission_spec.rb

```diff
@@ -0,0 +1,280 @@
+require 'rails_helper'
+
+module CCMS # rubocop:disable Metrics/ModuleLength
+  RSpec.describe Submission do
+    let(:legal_aid_application) { create :legal_aid_application, :with_applicant }
+
+    context 'Validations' do
+      it 'errors if no legal aid application id is present' do
+        sub = Submission.new
+        expect(sub).not_to be_valid
+        expect(sub.errors[:legal_aid_application_id]).to eq ["can't be blank"]
+      end
+    end
+
+    describe 'initial state' do
+      it 'puts new records into the initial state' do
+        sub = Submission.create(legal_aid_application: legal_aid_application)
+        expect(sub.aasm_state).to eq 'initialised'
+      end
+    end
+
+    describe '#process' do
+      context 'invalid state' do
+        it 'raises if state is invalid' do
+          sub = Submission.new(aasm_state: 'xxxxx')
+          expect {
+            sub.process!
+          }.to raise_error RuntimeError, 'Unknown state'
+        end
+      end
+
+      context 'initialised state' do
+        let(:submission) { create :submission, :initialised, legal_aid_application: legal_aid_application }
+        context 'operation successful' do
+          let(:response) { File.read(File.join(Rails.root, 'spec', 'services', 'ccms', 'data', 'reference_data_response.xml')) }
```

Comment:
Instead of `File.read(File.join(Rails.root, 'spec', 'services', 'ccms', 'data', 'reference_data_response.xml'))`

You could: `File.read(Rails.root.join 'spec', 'services', 'ccms', 'data', 'reference_data_response.xml')`

Though as the elements are all static I think this is cleaner:

```ruby
File.read(Rails.root.join 'spec/services/ccms/data/reference_data_response.xml')
```

Reply:
However, even better use
```ruby
cms_data_from_file 'reference_data_response.xml'
```
 as suggested in another comment.